File: Capstone/Capstone/Controllers/NoticiaController.cs
using Microsoft.AspNetCore.Mvc;
using Capstone.Models;

namespace Capstone.Controllers
{
    public class NoticiasController : Controller
    {
        private readonly ApplicationDbContext _context;

        public NoticiasController(ApplicationDbContext context)
        {
            _context = context;
        }

        

        [HttpPost]
        public IActionResult PublicarNoticia(long id)
        {
            var noticia = _context.Noticias.Find(id);
            if (noticia != null)
            {
                // Aquí puedes agregar lógica para marcar la noticia como publicada
                // Por ejemplo, agregar una columna `Estado` en la base de datos
                _context.SaveChanges();
            }
            return RedirectToAction("Admin_noticias");
        }

        [HttpPost]
        public IActionResult InsertarNoticiasPrueba()
        {
            var noticiasPrueba = new List<Noticias>
    {
            new Noticias
            {
                Titulo = "Primera Noticia",
                Noticia_id = 1,
                Detalle = "Este es un detalle de prueba.",
                Autor = "Admin",
                Fecha_publicacion = DateTime.UtcNow
            },
            new Noticias
            {
                Titulo = "Segunda Noticia",
                Noticia_id = 2,
                Detalle = "Este es otro detalle de prueba.",
                Autor = "Admin2",
                Fecha_publicacion = DateTime.UtcNow
            }
        };


            _context.Noticias.AddRange(noticiasPrueba);
            _context.SaveChanges();

            return RedirectToAction("Admin_noticias");
        }
        public IActionResult Admin_noticias()
        {
            var noticias = _context.Noticias.ToList();
            return View(noticias);
        }


        [HttpPost]
        public IActionResult RechazarNoticia(long id)
        {
            var noticia = _context.Noticias.Find(id);
            if (noticia != null)
            {
                // Lógica para rechazar la noticia (puede incluir eliminarla o marcarla como rechazada)
                _context.Noticias.Remove(noticia);
                _context.SaveChanges();
            }
            return RedirectToAction("Admin_noticias");
        }
    }
}

File: Proyecto Vecinos Unidos/controlador/noticia_controller.py
from flask import Flask, render_template, request, redirect, url_for, flash
from flask_sqlalchemy import SQLAlchemy
from datetime import datetime
from models import  noticias  # Asegúrate de importar correctamente tu modelo

app = Flask(__name__)
app.secret_key = 'clave_secreta'
app.config['SQLALCHEMY_DATABASE_URI'] = 'postgresql+pg8000://postgres:admin@localhost:5432/capstone'
db = SQLAlchemy(app)

@app.route('/noticias', methods=['GET', 'POST'])
def administrar_noticias():
    if request.method == 'POST':
        # Capturamos los datos del formulario para agregar una noticia
        titulo = request.form['titulo']
        detalle = request.form['detalle']
        autor = request.form['autor']
        fecha_publicacion = request.form['Fecha_publicacion']

        nueva_noticia = noticias(
            titulo=titulo,
            detalle=detalle,
            autor=autor,
            fecha_publicacion=datetime.strptime(fecha_publicacion, '%Y-%m-%d')
        )
        db.session.add(nueva_noticia)
        db.session.commit()
        flash('Noticia agregada exitosamente.', 'success')
        return redirect(url_for('administrar_noticias'))

    # Obtenemos todas las noticias para mostrarlas en la vista
    noticias = noticias.query.all()
    return render_template('noticias.html', noticias=noticias)

@app.route('/publicar/<int:noticia_id>', methods=['POST'])
def publicar_noticia(noticia_id):
    # Aquí puedes agregar lógica para cambiar el estado de la noticia a "publicada"
    flash(f'Noticia {noticia_id} publicada con éxito.', 'success')
    return redirect(url_for('administrar_noticias'))

@app.route('/rechazar/<int:noticia_id>', methods=['POST'])
def rechazar_noticia(noticia_id):
    # Aquí puedes agregar lógica para eliminar o marcar como rechazada la noticia
    flash(f'Noticia {noticia_id} rechazada.', 'danger')
    return redirect(url_for('administrar_noticias'))

if __name__ == '__main__':
    app.run(debug=True)
